vacuum-advisor/vacuum-simulator.mdx

```diff
@@ -0,0 +1,182 @@
+You can also create a permalink of your current simulation and modified settings
+to share with your team members.
+
+### Limitation
+
+* The current autovacuum count can be much bigger than the simulation count with
+  frequently autovacuumed tables, since the simulation can only perform
+  autovacuum every 10 minutes.
+
+## Triggered by dead rows
+
+To prevent table bloat, Postgres runs autovacuums periodically to clean up dead
+rows. You can learn more about bloat in the
+[What is bloat?](/docs/vacuum-advisor/what-is-bloat) page.
+
+![Triggered by dead rows](vacuum_simulator_dead_rows.png)
+
+When the number of dead rows exceeds the following threshold, dead rows
+autovacuums will be triggered:
+
+```
+threshold = autovacuum_vacuum_threshold + autovacuum_vacuum_scale_factor * number of tuples (pg_class.reltuples, number of live rows)
+```
+
+With VACUUM Simulator, you can tweak `autovacuum_vacuum_threshold` and
+`autovacuum_vacuum_scale_factor` in this section. Lowering these configuration
+settings usually triggers autovacuums more, as the threshold becomes lower and
+dead rows count will reach to the threshold more often.
+You can first try lowering `autovacuum_vacuum_scale_factor` to see how it will
+affect the behavior, before touching `autovacuum_vacuum_threshold`.
+
+On the other hand, if you are observing too many autovacuums with the table,
+it might be creating relatively many dead rows compares to the total live rows.
+In that case, making `autovacuum_vacuum_scale_factor` 0 and adjust
+`autovacuum_vacuum_threshold` to reasonable amount might be a better setting to
+have more control over the autovacuum behavior and lower the total autovacuum
+count.
+
+This section also contains the graph of avoidable growth, calculated based on
+the following factors:
+
+* How many reusable rows are available (already VACUUMed rows)
+* How many dead rows are there (to be VACUUMed rows)
+
+Let's say there are 10 reusable rows and 20 dead rows. With this situation, if
+30 rows are INSERTed:
+
+1. Use 10 reusable rows
+2. Use 20 totally new rows
+
+However, if 20 dead rows were already VACUUMed at this point, the reusable rows
+will be 30 rows and all INSERTs will use these reusable rows without acquiring
+space for new rows. We'll consider these 20 dead rows as avoidable growth.
+When the table is observing this pattern a lot, it indicates that autovacuums
+are not running frequent enough. Try tweaking configuration settings to trigger
+simulation autovacuums more and see if avoidable growth decreases.
+
+There is a specific insight for this:
+[Bloat - Insufficient VACUUM Frequency](/docs/checks/vacuum/insufficient_vacuum_frequency).
+
+### Limitation
+
+* The simulation data tends to trigger autovacuum earlier than the actual
+  triggered timing, especially with tables with lots of UPDATEs.
+  Occasionally, UPDATEs operations don't create dead rows, or even decrease dead
+  rows when "mini autovacuum" happened with these operations.
```

Comment:
I actually kinda forgot where I got this wording from, but I think I got it from our 5 mins postgres: https://pganalyze.com/blog/5mins-postgres-debug-UPDATE-bloated-tables-auto-explain-pageinspect#hot-pruning-in-postgres

> Postgres can actually do a “mini-VACUUM” of sorts, where it just vacuums the page that you're modifying, because it’s already modifying that page anyway.

I'll put the link to there